common/service/dynamicconfig/config.go

```diff
@@ -44,14 +48,31 @@ func NewCollection(client Client, logger log.Logger) *Collection {
 	}
 }
 
+// NewCollection creates a new collection
+func NewCollection(
+	client Client,
+	clusterName string,
+	logger log.Logger,
+) *Collection {
```

Comment:
How about changing the signature to `NewCollection(Client, log.Logger, ...Filter)`? I think it will make the code more generic and we don't need the separate `NewCollectionForTest()` function.

Reply:
Good point. Changed it